public/js/components/Editor.js

```diff
@@ -71,45 +71,53 @@ const Editor = React.createClass({
     }
   },
 
-  clearDebugLine(line) {
-    this.editor.codeMirror.removeLineClass(line - 1, "line", "debug-line");
-  },
-
-  setDebugLine(line) {
-    this.editor.codeMirror.addLineClass(line - 1, "line", "debug-line");
-    this.editor.alignLine(line);
-  },
-
-  setSourceText(newSourceText, oldSourceText) {
-    if (newSourceText.get("loading")) {
-      this.setText("Loading...");
-      return;
+  updateDebugLine(prevProps, nextProps) {
+    if (prevProps.selectedFrame) {
+      const line = prevProps.selectedFrame.location.line;
+      this.editor.codeMirror.removeLineClass(line - 1, "line", "debug-line");
     }
-
-    if (newSourceText.get("error")) {
-      this.setText("Error");
-      console.error(newSourceText.get("error"));
-      return;
+    if (nextProps.selectedFrame) {
+      const line = nextProps.selectedFrame.location.line;
```

Comment:
it would be nice to have a `debugLine` function or prop passed down from connect. We're going to have lots of locations and will want to be comparing them, so we should have language around debug lines, highlight lines, jump lines... With highlight lines / jump lines m guess is that at some point we don't know if a jump line is a highlight line, it might just be a debug line... *(might not make sense)*